### ISSUE TITLE
Fetch old predictors with old analysis

### ISSUE THREAD

**Issue by adelavega** [neuroscout/neuroscout#222]
Typically, when one fetches the predictors for a set of runs, the default is to get the latest version of the predictors.

However, if you clone an analysis, or edit an old analysis (that was never compiled), you could be using old predictor ids. 

However, a fresh fetch of predictors would give conflicting info on which predictors are selected. 

One solution is to always update predictors when cloning (although this might still be a problem for editing old analyses).

Another option is to give an override analysis id to the `/predictors` route that would override any similary named predictors, with the version used by that analysis.  So that if there are old predictors associated, it gives those rather than the latest ones

**Comment by tyarkoni** [comment]
I was thinking about this some more, and an alternative approach that' might be more robust and flexible is to add an API route that returns different versions for a given predictor. On the client side, for predictors that have multiple versions, we could display a "versions" button that pops up a modal dialog of available versions when clicked--and then the user can choose which one. It would always default to the latest, so that most of the time nobody would bother changing the version. But the option would be there if the user wanted it. And for cloned/edited analyses, we could still make the default the one that was previously run.

**Comment by adelavega** [comment]
Yeah, that would work. I mean that's already available because you can query `/api/predictors` and request all, rather than only the latest. Then it would be up to the front end to appropriately display that information (and you can infer order by id... not sure if I give timestamps). I would mainly worry that it may be a very large response if we update often (which maybe we wont), and potentially confusing to users unless its a bit of a hidden option (since 99% of people won't want to do this).

**Comment by adelavega** [comment]
I think given the changes to `fill` this is just a matter of either updating the analysis using `/fill` or ensuring to request the correct predictors on `/api/predictors` when showing the list. You can fetch by ID, so this would be the way to do it, in addition to all the latest available.
